Tests/Unit/LoRaTools/MacCommandTests.cs

```diff
@@ -0,0 +1,292 @@
+            ToBytes_Internal(new TxParamSetupRequest(dwellTimeSetting), new[] { expected });
+    }
+
+    public sealed class TxParamSetupAnswerTests : MacCommandTests<TxParamSetupAnswer>
+    {
+        public TxParamSetupAnswerTests() :
+            base(Cid.TxParamSetupCmd,
+                 new TxParamSetupAnswer(),
+                 Array.Empty<byte>())
+        { }
+    }
+
+    public sealed class DevStatusAnswerTests : MacCommandTests<DevStatusAnswer>
+    {
+        public DevStatusAnswerTests() :
+            base(Cid.DevStatusCmd,
+                 new DevStatusAnswer(1, 2),
+                 new byte[] { 1, 2 })
+        { }
+
+        [Fact]
+        public void FromBytes_Success() => FromBytesTest(actual => new { actual.Cid, actual.Battery, actual.Margin },
+                                                         bytes => new DevStatusAnswer(new ReadOnlySpan<byte>(bytes)));
+    }
+
+    public sealed class DevStatusRequestTests : MacCommandTests<DevStatusRequest>
+    {
+        public DevStatusRequestTests() :
+            base(Cid.DevStatusCmd,
+                 new DevStatusRequest(),
+                 Array.Empty<byte>())
+        { }
+
+        [Fact]
+        public void Deserializes_Correctly() =>
+            DeserializationTest(actual => new { actual.Cid }, "{cid:6}");
+    }
+
+    public sealed class DutyCycleAnswerTests : MacCommandTests<DutyCycleAnswer>
+    {
+        public DutyCycleAnswerTests() :
+            base(Cid.DutyCycleCmd,
+                 new DutyCycleAnswer(),
+                 Array.Empty<byte>())
+        { }
+    }
+
+    public sealed class DutyCycleRequestTests : MacCommandTests<DutyCycleRequest>
+    {
+        public DutyCycleRequestTests() :
+            base(Cid.DutyCycleCmd,
+                 new DutyCycleRequest(3),
+                 new byte[] { 3 })
+        { }
+
+        [Fact]
+        public void Deserializes_Correctly() =>
+            DeserializationTest(actual => new { actual.Cid, actual.DutyCyclePL }, "{cid:4,dutyCyclePL:3}");
+    }
+
+    public sealed class LinkAdrAnswerTests : MacCommandTests<LinkADRAnswer>
+    {
+        public LinkAdrAnswerTests() :
+            base(Cid.LinkADRCmd,
+                 new LinkADRAnswer(1, true, false),
+                 new byte[] { 0b110 })
+        { }
+
+        [Fact]
+        public void FromBytes_Success() => FromBytesTest(actual => new { actual.Cid, actual.PowerAck, actual.DRAck, actual.CHMaskAck },
+                                                         bytes => new LinkADRAnswer(new ReadOnlySpan<byte>(bytes)));
+    }
+
+    public sealed class LinkAdrRequestTests : MacCommandTests<LinkADRRequest>
+    {
+        private static readonly Func<LinkADRRequest, object> Transform = actual => new
+        {
+            actual.Cid,
+            actual.DataRate,
+            actual.TxPower,
+            actual.ChMask,
+            actual.ChMaskCntl,
+            actual.NbRep
+        };
```

Comment:
Curious about 2 things:

- Why is this exceptionally expressed differently than in all other tests?
- Why isn't is just a regular method, like below?

```suggestion
        private static object Transform(LinkADRRequest actual) => new
        {
            actual.Cid,
            actual.DataRate,
            actual.TxPower,
            actual.ChMask,
            actual.ChMaskCntl,
            actual.NbRep
        };
```

Reply:
What do you mean by expressed differently? I didn't inline it, since it has two references. All other "transform" methods have only a single reference. Will apply your suggestion though.